data-raw/workflow/workflow_union_3_years_in_one_table_pseudonymised.sql

```diff
@@ -0,0 +1,232 @@
+              when (len = 5.0 and substr(pcd_temp, 3, 1) = 'i') then (substr(pcd_temp, 1, 2) || '1' || substr(pcd_temp, 4, 2))
+              when (len = 5.0 and substr(pcd_temp, 3, 1) = 'l') then (substr(pcd_temp, 1, 2) || '1' || substr(pcd_temp, 4, 2))
+              when (len = 5.0 and substr(pcd_temp, 4, 1) = '5') then (substr(pcd_temp, 1, 3) || 's' || substr(pcd_temp, 5, 1))
+              when (len = 5.0 and substr(pcd_temp, 4, 1) = '0') then (substr(pcd_temp, 1, 3) || 'o' || substr(pcd_temp, 5, 1))
+              when (len = 5.0 and substr(pcd_temp, 5, 1) = '5') then (substr(pcd_temp, 1, 4) || 's')
+              when (len = 5.0 and substr(pcd_temp, 5, 1) = '0') then (substr(pcd_temp, 1, 4) || 'o')
+              else pcd_temp
+            end as pcd_temp
+          from (
+            select q01.*, length(postcode) as len, postcode as pcd_temp
+            from (
+              select
+                postcode_old,
+                upper(regexp_replace(postcode, '[^[:alnum:]]', '')) as postcode
+              from (
+                select
+                  postcode_old,
+                  case when (length(postcode) = 0.0) then null when not (length(postcode) = 0.0) then postcode end as postcode
+                from (
+                  select distinct postcode as postcode_old, postcode
+                  from dim.ons_postcode_data_dim
+                  where
+                    (country_code = 'E92000001') and
+                    (not((postcode is null)))
+                ) q01
+              ) q01
+            ) q01
+          ) q01
+        ) rhs
+          on (lhs.postcode_old = rhs.postcode_old)
+      ) q01
+    ) q01
+    where (rank = 1.0)
+) opdd
+on base.bsa_postcode = opdd.postcode;
+
+alter table int646_base_20200401_20210331_pseudo drop column bsa_sla;
+alter table int646_base_20200401_20210331_pseudo drop column match_sla;
+alter table int646_base_20200401_20210331_pseudo drop column match_sla_std;
+alter table int646_base_20200401_20210331_pseudo drop column nhs_no;
+alter table int646_base_20200401_20210331_pseudo rename column nhs_no_hash to nhs_no;
+commit;
+
+grant select on int646_base_20200401_20230331_pseudo to migar, adnsh, mamcp;
+
+create index int646_i01 on int646_base_20200401_20230331_pseudo (fy);
+create index int646_i02 on int646_base_20200401_20230331_pseudo (fy, year_month);
+
+create index int646_i03 on int646_base_20200401_20230331_pseudo (fy, age_band, gender);
+create index int646_i04 on int646_base_20200401_20230331_pseudo (fy, age_band, gender, pcd_region_code, pcd_region_name);
+create index int646_i05 on int646_base_20200401_20230331_pseudo (fy, age_band, gender, pcd_icb_code, pcd_icb_name);
+create index int646_i06 on int646_base_20200401_20230331_pseudo (fy, age_band, gender, pcd_lad_code, pcd_lad_name);
+create index int646_lsoa_i01 on int646_base_20200401_20230331_pseudo (fy, age_band, gender, lsoa_code);
```

Comment:
I've added a new index for `lsoa_code`, wherever there were already indices with a geo component